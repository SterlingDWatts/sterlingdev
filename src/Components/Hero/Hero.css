.Hero {
  background-color: #00685b;
  color: white;
  height: 100vh;
  min-height: 585px;
  padding-top: 52px;
  text-align: left;
}

.Hero.Section {
  padding: 52px 0 0;
}

#sterling {
  position: relative;
  top: -56px;
  visibility: hidden;
}

.Hero__container {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.Hero__header {
  height: 30%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Hero header {
  width: fit-content;
}

.Hero h1 {
  font-size: 40px;
  line-height: 1;
}

.Hero__image_container {
  width: 100%;
  height: 50%;
}

.Hero__image {
  width: 100%;
  height: 100%;
  background-image: url(https://sterlingdwatts.github.io/portfolio/images/italy.jpg);
  background-position: center;
  background-size: cover;
}

.Hero--blurb {
  height: 20%;
  min-height: 152px;
  padding: 16px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.Hero--large {
  display: none;
}

@media screen and (min-width: 350px) {
  .Hero.Section {
    padding: 56px 0 0;
  }
}

@media screen and (min-width: 500px) {
  .Hero__image {
    width: 80%;
    margin: 0 auto;
    box-shadow: 0 1px 10px 3px rgba(0, 0, 0, 0.15),
      0 1px 4px 1px rgba(0, 0, 0, 0.15);
  }
}

@media screen and (min-width: 900px) {
  .Hero__container {
    flex-direction: column;
    flex-wrap: wrap;
  }

  .Hero__header,
  .Hero__image_container,
  .Hero--blurb {
    width: 50%;
  }

  .Hero__header {
    height: 60%;
    order: 1;
  }

  .Hero--blurb {
    height: 40%;
    order: 2;
  }

  .Hero__image_container {
    height: 100%;
    order: 3;
  }

  .Hero__image {
    width: 100%;
    background-image: url(https://sterlingdwatts.github.io/portfolio/images/italy-big.jpg);
    box-shadow: none;
  }
}
