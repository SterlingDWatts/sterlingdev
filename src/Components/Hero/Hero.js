import React, { Component } from "react";
import Section from "../Section/Section";
import "./Hero.css";

class Hero extends Component {
  render() {
    return (
      <Section className="Hero" id="sterling">
        <div className="Hero__container">
          <div className="Hero__header">
            <header>
              <h1>
                Sterling <br className="Hero--large" />
                Watts
              </h1>
              <span>Full Stack Web Developer</span>
            </header>
          </div>
          <div className="Hero__image_container">
            <div className="Hero__image"></div>
          </div>
          <div className="Hero--blurb">
            <p>
              I am a passionate Full Stack Web Developer that enjoys making
              beautiful and intuitive websites as well as analyzing the data
              that I can get from them!
            </p>
          </div>
        </div>
      </Section>
    );
  }
}

export default Hero;
