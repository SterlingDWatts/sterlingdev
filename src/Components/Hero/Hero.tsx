import React from "react";
import Section from "../Section/Section";
import "./Hero.css";

const Hero = () => {
  return (
    <Section className="Hero">
      <div id="sterling"> </div>
      <div className="Hero__container">
        <div className="Hero__header">
          <header>
            <h1>
              Sterling <br className="Hero--large" />
              Watts
            </h1>
            <span>Software Developer</span>
          </header>
        </div>
        <div className="Hero__image_container">
          <div className="Hero__image"></div>
        </div>
        <div className="Hero--blurb">
          <p>
            I am a passionate Software Developer that enjoys making beautiful and intuitive websites as well as
            analyzing the data that I can get from them!
          </p>
        </div>
      </div>
    </Section>
  );
};

export default Hero;
