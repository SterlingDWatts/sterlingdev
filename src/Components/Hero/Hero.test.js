import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Hero from "./Hero";

describe("Hero Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Hero />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Hero />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
