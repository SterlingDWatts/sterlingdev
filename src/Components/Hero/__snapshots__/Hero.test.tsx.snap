// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders the UI as expected 1`] = `
<section
  className="Hero Section"
>
  <div
    id="sterling"
  >
     
  </div>
  <div
    className="Hero__container"
  >
    <div
      className="Hero__header"
    >
      <header>
        <h1>
          Sterling 
          <br
            className="Hero--large"
          />
          Watts
        </h1>
        <span>
          Software Developer
        </span>
      </header>
    </div>
    <div
      className="Hero__image_container"
    >
      <div
        className="Hero__image"
      />
    </div>
    <div
      className="Hero--blurb"
    >
      <p>
        I am a passionate Software Developer that enjoys making beautiful and intuitive websites as well as analyzing the data that I can get from them!
      </p>
    </div>
  </div>
</section>
`;
